feat: dynamically rendered pages for groups, urbit ids, applications

matildepark opened the pull request:
- [x] Pulls information from network explorer for Urbit ID ownership and sponsors if any
- [x] Basic markdown submission scaffold
- [x] Fallback basic landing pages for unsubmitted group links, etc.
- [x] Ship validation, otherwise 404s
- [x] Dynamic Twitter card generation
- [x] Guides for joining a group, getting an urbit ID
- [x] Linking all planet names on the blog, etc. to /id/ pages
- [x] Interactive submission form for submitting IDs, groups, applications to the directory
- [x] Still need a few more Urbit IDs and groups for initial population
- [x] Custom 404 page

<img width="866" alt="Screen Shot 2022-04-22 at 2 19 36 PM" src="https://user-images.githubusercontent.com/20846414/164796270-d1915c67-3dd7-43b9-b00d-1263893a06ba.png">

<img width="800" alt="Screen Shot 2022-04-22 at 1 23 51 PM" src="https://user-images.githubusercontent.com/20846414/164796322-32d41098-ad72-49a5-9e3b-4a0077dc51ac.png">

matildepark committed:
build: add prettierrc

matildepark committed:
components: addCopyLink

matildepark committed:
Meta: allow disabling fallback image

matildepark committed:
packages: add axios, sigil-js, urbit-ob

matildepark committed:
components: add sigil component

matildepark committed:
pages: add dynamic route for groups

matildepark committed:
pages: add dynamic routes for urbit ids

matildepark committed:
content: add initial group info

matildepark committed:
content: add initial id info

matildepark committed:
pages/id: don't send invalid patps to network explorer

matildepark commented:
Note for voyeurs: 500s on Vercel deployment, need to refactor to pull some fetching out of the server-side and use SWR, I think.

matildepark committed:
group,id: content / markdown fallback tweak

matildepark committed:
id: remove dangling import

matildepark committed:
markdown: allow for disabling plugins in SSR pages

gordoncc commented:
Looking pretty, pretty good.  Already polishing with fine grit here:

- [x] Update “Copy Link” background color to #24221E
- [x] Add “-webkit-font-smoothing: antialiased;” to .font-bold CSS class
- [x] If possible make rows of 2 for metadata on mobile (see mobile design)
- [x] Add “View on Network Explorer” link for Group ID pages (see design)
- [x] Change class on “Learn how to get your Urbit ID” from “text-sm” to “text-md”
- [x] Remove “Parent” metadata from galaxies

matildepark committed:
markdown: tweak optional plugin construction

matildepark committed:
id: hide parent on galaxy pages

matildepark committed:
group, id, globals.css: address @gordoncc review (#1418)

matildepark committed:
pages, components: wrap all mentions of planets in /id/ links

matildepark committed:
Merge branch 'master' into mp/groups

matildepark commented:
All comments addressed @gordoncc, just need to generate cards, make the fallback 404 pages, add the guides now.

matildepark committed:
pages: add next/link imports

matildepark committed:
cache: don't index id pages

matildepark committed:
sigil: enable icon sigils

matildepark committed:
search: add id searching

matildepark commented:
You can search for Urbit IDs now (including just searching numbers, automatically converting to patps).

<img width="1099" alt="Screen Shot 2022-04-24 at 4 12 28 PM" src="https://user-images.githubusercontent.com/20846414/165000678-fb93cc59-c193-4057-a200-0fec3376fd7f.png">

<img width="984" alt="Screen Shot 2022-04-24 at 4 12 48 PM" src="https://user-images.githubusercontent.com/20846414/165000680-c553c16c-b412-4502-9111-90b7570b76d3.png">

matildepark committed:
pages/id: most basic of basic urbit id og cards

matildepark commented:
Deployed separate Vercel repository for generating OpenGraph cards at https://github.com/matildepark/urbit-id-og-cards.

Hooking that into this and testing it with the Twitter card generator.

matildepark committed:
id: skip metadata block

matildepark commented:
Currently needs more twitter card metadata for it to render, but the metadata so far is correct; I need to prevent really long patps from crashing the sigil renderer.

matildepark committed:
id: full metadata

matildepark committed:
group: test group cards

matildepark committed:
group, id: aggressively cache

Pages are considered stale after 1 hour,
but for a week afterward will be served stale + refetched in background
when requested.

matildepark committed:
search, sigil: no comets, no moons

matildepark committed:
id: no comets, no moons

matildepark committed:
pages, components: recomponentize public gateway pages

matildepark committed:
pages/group, id: add 404 fallbacks

matildepark committed:
pages: add application page

matildepark committed:
lib: loosen getPostSlugs for non-existent dirs, loosen getAllPosts for non-date sorting

matildepark committed:
pages: specify sort

matildepark committed:
pages/id: show all known groups, applications on id pages

matildepark committed:
pages: add submission forms and dynamic PR construction

matildepark committed:
content/id: add haddef-sigwen

Because of course.

matildepark committed:
pages: add group + app guide content and template, links to guides on pages

matildepark commented:
All features on place, shippable — only thing left is "more content", but that isn't blocking.

matildepark commented:
I actually think the only thing left is for "get an Urbit ID" to link somewhere.

matildepark committed:
id: amend groups/apps for mocks

matildepark committed:
id: add getting an urbit id guide

jalehman commented on content/application/guide.md:
```diff
@@ -0,0 +1,39 @@
+
+However, we can get started a simple app from Urbit developer [`~paldev`](/id/~paldev).
+
+## 3. Search for a Developer
```
You're not really "Search"ing for a developer; rather, you're entering a known one. 

You can also enter an entire app shortcode into this search bar, much as you can with groups, e.g. ~paldev/pals can be pasted directly from one of these pages.

matildepark replied:
That actually doesn't work right now — I put in a fix so that it does at https://github.com/urbit/urbit/pull/5738 though.

gordoncc replied:
We can switch up the copy to refer to shortcodes once they're working.

jalehman commented on content/application/guide.md:
```diff
@@ -0,0 +1,39 @@
+
+For long-term use, you’ll want to get a permanent Urbit ID and consider a running your urbit in the cloud or on a hosted service.
+
+Read our [Getting Started guide](https://urbit.org/install) for details.
```
```suggestion
Read our [Getting Started guide](https://urbit.org/getting-started) for details.
```

/install redirects to /getting-started

jalehman reviewed (commented):
* In preparation for eventual directories/indexes urls should probably be plural, e.g. urbit.org/group -> urbit.org/groups
* Not sure if this is viable, but guides could be written to accept query parameters that reference the group/app so that the guide customizes itself to instruct you to join/install *that* particular group/app.
* We need a sample application, e.g. Groups, to test with. I bet you could write the page for this @matildepark

matildepark commented:
> Not sure if this is viable, but guides could be written to accept query parameters that reference the group/app so that the guide customizes itself to instruct you to join/install that particular group/app.

Probably viable, it's just queries, we all know and love those. Just need the appropriate copy to substitute.

matildepark committed:
pages: pluralise group, id, application links

matildepark committed:
components: add ResourceCard, GatewayImage

matildepark committed:
GatewayImage: allow for background colours

matildepark committed:
pages/applications: render dev as link if it's a patp

matildepark committed:
content/applications: add ~paldev/suite

matildepark committed:
ResourceCard, Image: wrapping fix

matildepark committed:
pages/ids: fix broken import for plural

matildepark committed:
components/Contact: change UC CTA to group link

Commit:
Delete cliff.md

~palfun-foslup recommended not including this one since it is dangerous for your ship if you poke around without knowing what you're doing

matildepark committed:
groups/submit: remove channels field

matildepark committed:
pages/ids: add nicknames, sigil colors

matildepark committed:
pages/applications: add layered bg+image render

matildepark committed:
pages/groups: add title to submit page

matildepark committed:
pages/ids: fallback to wall-600 not black

matildepark committed:
id: dynamic twitter card params

matildepark committed:
content/ids: update haddef-sigwen

matildepark committed:
id: remove console.log

matildepark committed:
pages/application: send params for stacking colors + images to dynamic card generator

matildepark committed:
lib/cache: fix dropped plural folder

matildepark committed:
content/applications: delete dummy.md

gordoncc reviewed (commented):
**Design/Content**

- [x] Add 5px below the @p under the nickname on ID pages
- [x] Reduce space below Page title headings on guides “Joining groups on Urbit”, “Getting an Urbit ID” and “Installing applications on Urbit” - see designs
- [ ] remove links to developers on application page if it's an org or moon (unless it is a planet @p and it's automatically linked)

**Features**

- [x] Add color for groups pages and submission (background color in image tile option instead of image)


**URL Updates**

- [x] Change /ids/get to “/guides/getting-an-urbit-id”
- [x] Change /groups/guide to “/guides/joining-groups”
- [x] Change /groups/guide to “/guides/installing-applications”

Commit:
added initial solicited groups, fixed typo in group page

matildepark committed:
pages/groups: make color/image submission more clear

matildepark committed:
Merge branch 'mp/groups' of github.com:urbit/urbit.org into mp/groups

matildepark committed:
content/groups: update to use tile property

matildepark committed:
content/groups add door.link change

matildepark commented:
Setting a colour for groups was already supported — the Image component written check for hex codes and renders it as a background — so I made it more clear by renaming the markdown property from `image` to `tile`.

matildepark committed:
pages/id: use data.tile for group cards

matildepark commented:
> remove links to developers on application page if it's an org or moon (unless it is a planet @p and it's automatically linked)

I figured it was a nice touch rather than making you type it out (given often times it's a moon hosting the application with an original developer being an unrelated planet). What behaviour is best here?

matildepark committed:
GatewayHeader: add pb-1 to subtitles

matildepark committed:
pages/**/guide: use short sections

matildepark committed:
content/guides: rename routes for guide pages

matildepark committed:
pages/guides: filter out special cased app page

Commit:
added initial applications

matildepark committed:
components, pages: MetadataBlocks always take basis-1/2

Commit:
polishing guides

gordoncc reviewed (approved):
I think we're solid for an mvp soft launch.

matildepark commented:
May I hit the button, @jalehman?

Commit:
fixes on new york group

matildepark committed:
pages/groups: update data.image -> data.tile

matildepark committed:
Merge branch 'mp/groups' of github.com:urbit/urbit.org into mp/groups

matildepark committed:
pages/groups: option chain

matildepark committed:
components/gateway/Image: update placeholder

matildepark committed:
pages/ids: loosen for desig'd input
